backend/src/match/match.gateway.ts

```diff
@@ -25,11 +26,33 @@ export class MatchGateway {
   ) {}
 
   @SubscribeMessage("find")
-  handleFind(@ConnectedSocket() client: Socket): void {
+  async handleFind(
+    @ConnectedSocket() client: Socket,
+    @MessageBody() difficultyLevel: string,
+  ): Promise<void> {
     this.logger.info(`Handling find match request: ${client.id}`);
     const userId = Number(client.handshake.headers.authorization);
-    // TODO: Call service.
-    return;
+    const match = await this.service.searchMatch(
+      userId,
+      difficultyLevel,
+      client.id,
+    );
+
+    if (!match) {
+      return;
+    }
+
+    // Get sockets by ID and let them join the same room
+    const sockets = await this.server.fetchSockets();
```

Comment:
We should be able to use `this.sever.sockets.sockets.get(...)` to retrieve the client socket instances rather than iterating over all sockets.

Reply:
I wanted to do this initially, but I'm not sure why `this.sever.sockets.sockets` is undefined

Reply:
Try this instead, replace the type of `server` and import `Namespace` from `socket.io`.
```
  server: Namespace;
```
And then use:
```    
    this.server.sockets.get(match.result[0].socketId)?.join(match.roomId);
    this.server.sockets.get(match.result[1].socketId)?.join(match.roomId);
```